cinema/serializers.py

```diff
@@ -68,13 +78,76 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+    def validate(self, attrs):
+        if not (1 <= attrs["row"] <= attrs["movie_session"].cinema_hall.rows):
+            raise serializers.ValidationError(
+                {
+                    "seat":
+                        f"row must be in range "
+                        f"[1, {attrs['movie_session'].cinema_hall.rows}], "
+                        f"not {attrs['row']}"
+                }
+            )
+        if not (
+                1 <= attrs["seat"] <= attrs["movie_session"]
+                .cinema_hall.seats_in_row
+        ):
+            raise serializers.ValidationError(
+                {
+                    "seat":
+                        f"seat must be in range "
+                        f"[1, {attrs['movie_session'].cinema_hall.rows}], "
+                        f"not {attrs['seat']}"
+                }
+            )
+
+
+class TakenPlaceSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TakenPlaceSerializer(
+        many=True, read_only=True, source="tickets"
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=False, allow_empty=False)
+
+    def create(self, validated_data):
+        with transaction.atomic():
+            tickets_data = validated_data.pop("tickets")
+            order = Order.objects.create(**validated_data)
```

Comment:
Use transaction.atomic as decorator instead of context mananger. It makes your code more readable